File: Front/ProEventos-App/src/app/models/Event.ts
import { Batch } from "./Batch";
import { SocialNetwork } from "./SocialNetwork";
import { Speaker } from "./Speaker";

export interface Event {
    id: number;
    location: string;
    eventDate?: Date;
    subject: string;
    peopleQuantity: number;
    imageURL: string;
    phone: string;
    email: string;
    batch: Batch[];
    socialNetworks: SocialNetwork[];
    speakersEvents: Speaker[];
}

File: Back/src/ProEventos.Domain/Event.cs
using System;
using System.Collections.Generic;

namespace ProEventos.Domain
{
    public class Event
    {
        public int Id { get; set; }
        public string Location { get; set; }
        public DateTime? EventDate { get; set; }
        public string Subject { get; set; }
        public int PeopleQuantity { get; set; }
        public string ImageURL { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public IEnumerable<Batch> Batches { get; set; }
        public IEnumerable<SocialNetwork> SocialNetworks { get; set; }
        public IEnumerable<SpeakerEvent> SpeakersEvents { get; set; }
    }
}